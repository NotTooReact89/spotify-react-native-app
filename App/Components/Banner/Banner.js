// @flow

import React from 'react'
import { View, TouchableHighlight } from 'react-native'
import Icon from '../Icon'
import Text from '../Text'
import Styles from './BannerStyles'
import { Colors } from '../../Styles'
import { TestProps, CustomAnimations } from '../../Lib'

const Animatable = require('react-native-animatable')

const { slideOutUp } = CustomAnimations

Animatable.initializeRegistryWithDefinitions({
  slideOutUp,
})

const animationDuration = 1000

export type Props = {
  text: string,
  onPress: () => void,
  testId?: string,
}

class Banner extends React.Component<Props, *> {
  props: Props
  banner: ?Animatable.View

  handleDismiss = () => {
    if (this.banner) {
      this.banner.slideOutUp(animationDuration).then(this.props.onPress)
    }
  }

  render() {
    const { text } = this.props
    const testId = this.props.testId ? `${this.props.testId}_` : ''

    return (
      <Animatable.View
        ref={input => {
          this.banner = input
        }}
        duration={animationDuration}
        animation="slideInDown"
        {...TestProps(`${testId}banner_container`)}
      >
        <View style={Styles.container}>
          <View style={Styles.contentContainer}>
            <Icon type="infoInverse" />
            <Text
              inverse
              style={Styles.text}
              {...TestProps(`${testId}banner_txtMessage`)}
            >
              {text}
            </Text>
          </View>
          <TouchableHighlight
            style={Styles.iconContainer}
            onPress={this.handleDismiss}
            underlayColor={Colors.base.transparent}
            {...TestProps(`${testId}banner_btnClose`)}
          >
            <Icon type="closeInverse" />
          </TouchableHighlight>
        </View>
      </Animatable.View>
    )
  }
}

export default Banner
