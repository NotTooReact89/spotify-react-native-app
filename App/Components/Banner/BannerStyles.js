// @flow

import { StyleSheet } from 'react-native'
import { Colors, Spaces } from '../../Styles'

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.base.info,
    flexDirection: 'row',
    padding: Spaces.md,
    zIndex: 10,
  },

  contentContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },

  text: {
    marginLeft: Spaces.md,
    flex: 1,
  },

  iconContainer: {
    paddingLeft: Spaces.lg,
  },
})
