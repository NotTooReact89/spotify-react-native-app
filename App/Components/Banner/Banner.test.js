// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Banner from './Banner'

it('should render banner correctly', () => {
  const tree = renderer
    .create(
      <Banner text="some text" onPress={jest.fn()} testId="test_screen" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
