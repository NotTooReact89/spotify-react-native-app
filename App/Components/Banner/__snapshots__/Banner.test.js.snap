// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render banner correctly 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
  testID="test_screen_banner_container"
>
  <View
    style={
      Object {
        "backgroundColor": "#008392",
        "flexDirection": "row",
        "padding": 20,
        "zIndex": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "resizeMode": "contain",
            },
            Object {
              "height": 24,
              "width": 24,
            },
            Object {},
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Lato-Light",
              "fontSize": 15,
              "lineHeight": 22.5,
            },
            Object {
              "flex": 1,
              "marginLeft": 20,
            },
          ]
        }
        testID="test_screen_banner_txtMessage"
      >
        some text
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "paddingLeft": 40,
        }
      }
      testID="test_screen_banner_btnClose"
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "resizeMode": "contain",
            },
            Object {
              "height": 24,
              "width": 24,
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
</View>
`;
